compare_platforms/compare_platforms_overflow.c

```diff
@@ -96,27 +96,34 @@ int main(int argc, char *argv[])
 
     assert_positions(publicreview, permissions, 0x20, 0x1001);
 
-    /*#ifdef __CHERI_PURE_CAPABILITY__
-    #if (defined(__aarch64__))
-        printf("\nOn morello-purecap, you may reach this line, then the attempt to overflow by 1 may
-    " "cause a SIGPROT.\n"); #endif #endif*/
+#ifdef __CHERI_PURE_CAPABILITY__
+#if (defined(__aarch64__))
+            printf("\nOn morello-purecap, you may reach this line, then the attempt to overflow by 1 may
+        " "cause a SIGPROT.\n"); 
+#endif
+#endif
 
     overflow_publicreview(0x1001 + 1, 1, &review);
 
     const size_t oversz = review.permissions - review.publicreview + 2 - 0x1001;
 
-    /*#if (defined(__arm64__) && defined(__APPLE__))
+#if (defined(__arm64__) && defined(__APPLE__))
         printf("\nOn Apple arm64, you may reach this line, then the attempt to overflow by %zx may "
                "cause a segmentation fault.\n",
                oversz);
-    #endif*/
-    /*#if (defined(__x86_64__) && defined(__APPLE__))
+#endif
+
+#if (defined(__x86_64__) && defined(__APPLE__))
         printf("\nOn Apple x86_64, you may reach this line, then the attempt to overflow by %zx may
-    " "cause a segmentation fault.\n", oversz); #endif*/
-    /*#ifdef __CHERI_PURE_CAPABILITY__
-    #if (defined(__riscv))
-        printf("On riscv64-purecap, you may reach this line, then the attempt to overflow by %zx may
-    " "cause a SIGPROT.\n", oversz); #endif #endif*/
+    " "cause a segmentation fault.\n", oversz); 
+#endif
+
+#ifdef __CHERI_PURE_CAPABILITY__
+#if (defined(__riscv))
+            printf("On riscv64-purecap, you may reach this line, then the attempt to overflow by %zx may
+        " "cause a SIGPROT.\n", oversz); 
+#endif
+#endif
```

Comment:
Would it be possible to collapse these `if`s somehow, to not have these double `#endif`s?

Reply:
Fixed in 52b1253abd70afe77bca002e3da353716c28c960.